### DISCUSSION TITLE
Html processing

### DISCUSSION THREAD

**Discussion by vkosovskikh** [mdx-editor/editor#358] [Q&A]
Hello!

What exactly does the `suppressHtmlProcessing` prop do, and is it possible to make it so that HTML tags are not interpreted? In other words, can the source `<div>hello</div>` display the text `<div>hello</div>` instead of just the text `hello`?

**Comment by petyosi** [comment]
The property might have been named better, but in a nutshell, no - it won't treat HTML as plain text. You can do that on your own, but it needs a custom import/export visitor for the purpose.

**Comment by Mohsen2083** [comment]
Hi.
The suppressHtmlProcessing prop is not a standard prop in HTML or any particular framework/library, so its exact behavior would depend on the context in which it is used. However, based on the name, it suggests a feature that disables or suppresses the processing of HTML tags within a component or element.

If suppressHtmlProcessing is implemented as a prop in a custom component or library, it would likely be used to prevent HTML tags from being interpreted or rendered as markup. This means that any HTML tags within the content of the component or element would be treated as plain text and displayed as is, rather than being rendered as HTML elements.

For example, if you have the source <div>hello</div> and the suppressHtmlProcessing prop is enabled, the text displayed would indeed be <div>hello</div> rather than the interpreted HTML element. This can be useful in cases where you want to display HTML code or raw content that includes HTML tags without the tags being processed.

However, it's important to note that the behavior and existence of the suppressHtmlProcessing prop would depend on the specific library or component you are using. It is not a standard HTML or JavaScript feature, so its availability and behavior would vary depending on the implementation.

**Reply by petyosi** [reply]
No need to spam with AI bullshit.

**Comment by lukajincharadze** [comment]
The suppressHtmlProcessing prop is typically used in contexts where you want to display HTML content as plain text instead of rendering it as HTML.

When suppressHtmlProcessing is set to true, the component will treat the content as plain text and display any HTML tags as literal text rather than interpreting them as markup. This means that <div>hello</div> will be displayed as <div>hello</div> instead of being rendered as a div element with the text "hello".


might use suppressHtmlProcessing in a component:


`import React from 'react';

const MyComponent = ({ content, suppressHtmlProcessing }) => {
  return (
    <div>
      {suppressHtmlProcessing ? content : <div dangerouslySetInnerHTML={{ __html: content }} />}
    </div>
  );
};

export default MyComponent;
`
